### ISSUE TITLE
Decimal property is not displayed correctly in Variables in debugger

### ISSUE THREAD

**Issue by weinand** [OmniSharp/omnisharp-vscode#632]
_From @bgrainger on July 29, 2016 1:16_

- VSCode Version: Version 1.3.1 (1.3.1) e6b4afa53e9c0f54edef1673de9001e9f0f547ae
- OS Version: Mac OS X 10.10.5

#### Steps to Reproduce:

Create a .NET Core Console application with the following code:

```
namespace ConsoleApplication
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var t = new Test { Value = 1.0m };
            // SET BREAKPOINT HERE
            System.Console.WriteLine(t.Value);
        }
    }

    public class Test
    {
        public decimal Value
        {
            get { return _value; }
            set { _value = value;}
        }

        private decimal _value;
    }
}
```

Debug it by pressing F5. Break on the `Console.WriteLine` line and look at the Variables pane. The value of `Value` is displayed incorrectly (but `_value`) is correct. The incorrect value changes each time the program is run.

#### Actual Output:

<img width="215" alt="screenshot 2016-07-28 18 11 56" src="https://cloud.githubusercontent.com/assets/188129/17234605/483c03a6-54ef-11e6-9175-91fe95d4bdd9.png">

#### Expected Output:

Both `Value` and `_value` should show `1.0 [decimal]`.

_Copied from original issue: Microsoft/vscode#9929_

**Comment by weinand** [comment]
_From @sjones204g on July 29, 2016 2:10_

I'm getting this problem as well. VS Code 1.3.1 on Mac OSX 10.11.6.

**Comment by weinand** [comment]
The formatting of these values is done in the debug adapter of the C# extension.

**Comment by gregg-miskelly** [comment]
@noahfalk @Mikem8361 I _think_ this issue is a CoreCLR issue, but before I port this over, I wanted to know if you had a quick way to confirm/deny (ex: with mdbg)? The issue is that if we evaluate a function that returns a System.Decmial, unbox the resulting value, and then look at what memory backs the resulting value, the lower 64-bits are fine, but the high 64-bit look corrupt.

**Comment by mikem8361** [comment]
We don’t have a test specifically for this scenario, but I’m currently trying to repro with mdbg.  I’ll let you know the results.

**Comment by mikem8361** [comment]
Looks like I can repro the problem with mdbg on OS X and on Linux but not Windows. So it mostly likely a bug in coreclr.

mikem

**Comment by gregg-miskelly** [comment]
I opened https://github.com/dotnet/coreclr/issues/6935 to track this issue.

**Comment by gregg-miskelly** [comment]
The CoreCLR team has checked in a fix for their 1.1 release

**Comment by mayconbeserra** [comment]
H there, 

When will this fix be available?

Cheers,

**Comment by mikem8361** [comment]
The fix for properties returning decimal (or any other 16 byte value type) did making it into .NET Core 1.1. 

The second fix (dotnet/coreclr#7115) for funceval of functions passing in enregistered 16 byte value types didn't make it into 1.1.  It was too late (already past escrow) when I realized that.
